### PR TITLE
JESI-5315

### PR DESCRIPTION
**Pull request by mattfennerjesi**
feature: Simple filtering functions for collections of maps

### CODE REVIEW

File: src/io/jesi/backpack/collection.cljc [hunk 1]
```diff
@@ -371,3 +371,19 @@
   [m]
   (= (count m)
      (count (set (vals m)))))
+
```
Comment (mattfennerjesi) [inline src/io/jesi/backpack/collection.cljc:374]: wanted to create some helper methods for doing these filter operations, cause I find myself doing them quite often

File: src/io/jesi/backpack/collection.cljc [hunk 2]
```diff
@@ -371,3 +371,19 @@
...
+(defn filter-by
+  "Filters a collection where a key matches a predicate
+  e.g.
+  (let [m [{:id 1} {:id 2}]
+    (filter-by m :id (bp/p= 1))) ; returns {:id 1}"
```
Comment (axrs) [inline src/io/jesi/backpack/collection.cljc:379]:
```suggestion
    (filter-by m :id (bp/p= 1))) ; returns '({:id 1})"
```

File: src/io/jesi/backpack/collection.cljc [hunk 3]
```diff
@@ -371,3 +371,19 @@
...
+  [key-fn pred coll]
+  (filter (comp pred key-fn) coll))
+
+(defn filter-by=
+  "Filters a collection where a key matches a value
+  e.g.
+  (let [m [{:id 1} {:id 2}]
+    (filter-by= m :id 1)) ; returns {:id 1}"
```
Comment (axrs) [inline src/io/jesi/backpack/collection.cljc:387]:
```suggestion
    (filter-by= m :id 1)) ; returns '({:id 1})"
```

File: src/io/jesi/backpack/collection.cljc [hunk 4]
```diff
@@ -371,3 +371,19 @@
...
+
+(defn filter-by
+  "Filters a collection where a key matches a predicate
+  e.g.
+  (let [m [{:id 1} {:id 2}]
+    (filter-by m :id (bp/p= 1))) ; returns {:id 1}"
+  [key-fn pred coll]
+  (filter (comp pred key-fn) coll))
+
+(defn filter-by=
```
Comment (axrs) [inline src/io/jesi/backpack/collection.cljc:383]:
```suggestion
(defn filter-key=
```

Would `filter-key=` be a better name?

File: test/io/jesi/backpack/collection_test.cljc [hunk 5]
```diff
@@ -871,6 +871,50 @@
...
+
+      ; matches clojure.core/filter
+      (testing "returns empty seq when coll is nil"
+        (is= () (filter-by :id (constantly true) nil)))
+
+      (testing "can return 1 result"
+        (is= [one] (filter-by :id (bp/p= 1) coll)))
+
+      (testing "can return multiple results"
+        (is= [two three] (filter-by :id (bp/partial-right > 1) coll))
```
Comment (axrs) [inline test/io/jesi/backpack/collection_test.cljc:897]:
```suggestion
        (is= [two three] (filter-by :id (partial < 1) coll))
```

Our convention is to always try and use `<` where possible.

**Review by axrs — CHANGES_REQUESTED** [review]
You'll also have to re-run docs. If you can't I'll run em for you.

File: test/io/jesi/backpack/collection_test.cljc (hunk 5 above)
Comment (mattfennerjesi) [reply test/io/jesi/backpack/collection_test.cljc:897]: I found this confusing when you are finding the items which are > 1.
I might just use a different predicate.

File: src/io/jesi/backpack/collection.cljc (hunk 4 above)
Comment (mattfennerjesi) [reply src/io/jesi/backpack/collection.cljc:383]: yeah, was struggling with naming this one. done

File: test/io/jesi/backpack/collection_test.cljc (hunk 5 above)
Comment (mattfennerjesi) [reply test/io/jesi/backpack/collection_test.cljc:897]: changed

File: src/io/jesi/backpack/collection.cljc (hunk 3 above)
Comment (mattfennerjesi) [reply src/io/jesi/backpack/collection.cljc:387]: changed

File: src/io/jesi/backpack/collection.cljc (hunk 2 above)
Comment (mattfennerjesi) [reply src/io/jesi/backpack/collection.cljc:379]: changed

**Review by axrs — APPROVED** [review]

**Review by axrs — DISMISSED** [review]

**Review by axrs — APPROVED** [review]

**Pull request merged**